pyccl/halos/profiles/einasto.py

```diff
@@ -105,3 +113,30 @@ def _real(self, cosmo, r, M, a):
         if np.ndim(M) == 0:
             prof = np.squeeze(prof, axis=0)
         return prof
+
+    def _projected_integrate(self, cosmo, r, M, a):
+        r_use = np.atleast_1d(r)
+        M_use = np.atleast_1d(M)
+
+        # Comoving virial radius
+        R_M = self.mass_def.get_radius(cosmo, M_use, a) / a
+        c_M = self.concentration(cosmo, M_use, a)
+        R_s = R_M / c_M
+
+        alpha = self._get_alpha(cosmo, M_use, a)
+
+        def integrand(z):
+            x = np.sqrt(z**2. + r_use[None, :]**2.) / R_s[:, None]
+            return np.exp(-2. * (x**alpha[:, None] - 1.) / alpha[:, None])
+        prof = quad_vec(integrand, 0., np.inf)[0]
```

Comment:
Hmm is this accurate if we use `truncated=True`?

Reply:
Exactly the same as `truncated=False`

Reply:
humm, but shouldn't the range of integration be different here? Or shouldn't the integrand be truncated?

Reply:
The whole profile is obtained and then truncated just like all the other profiles. And yes, I should make the calculation more efficient by not doing integration for r_use > R_M.

Reply:
Fixed.

Reply:
I could not get `truncated` to work due to the array shapes of the broadcasted array and `r_use`. All the other profiles of different halo model  do not support `truncated` either, so this implementation is consistent with the current settings.

Reply:
sorry, to clarify, are you suggesting we disable the truncated option for einasto altogether? Not that I'd be opposed.

Reply:
Nevermind, I see what you did. Good for me.